media-gfx/iscan/iscan-3.62.0.ebuild

```diff
@@ -0,0 +1,83 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools desktop flag-o-matic udev xdg-utils
+
+DESCRIPTION="EPSON Image Scan v3 for Linux"
+HOMEPAGE="https://support.epson.net/linux/en/imagescanv3.php"
+SRC_URI="https://support.epson.net/linux/src/scanner/imagescanv3/common/imagescan_${PV}.orig.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+IUSE="gui"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="
+	dev-libs/boost:=
+	media-gfx/sane-backends
+	media-libs/tiff
+	virtual/libusb:1
+	virtual/jpeg
+	gui? ( dev-cpp/gtkmm:2.4 )
+	|| ( media-gfx/imagemagick[cxx] media-gfx/graphicsmagick[cxx] )
+"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/utsushi-0.$(ver_cut 2-3)"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-3.61.0-ijg-libjpeg.patch
+	"${FILESDIR}"/${PN}-3.61.0-imagemagick-7.patch
+)
+
+src_prepare() {
+	default
+
+	# Remove vendored libraries
+	rm -r upstream/boost || die
+	# Workaround for deprecation warnings:
+	# https://gitlab.com/utsushi/utsushi/issues/90
+	sed -e 's|-Werror||g' -i configure.ac || die
+	eautoreconf
+}
+
+src_configure() {
+	# Workaround for:
+	# /usr/lib64/utsushi/libutsushi.so.0: undefined symbol: libcnx_usb_LTX_factory
+	append-ldflags $(no-as-needed)
+	local myconf=(
+		$(use_with gui gtkmm)
+		--enable-sane-config
+		--enable-udev-config
+		--with-boost=yes
+		--with-jpeg
+		--with-magick
+		--with-magick-pp
```

Comment:
`configure.ac` contains:
```
AS_IF([test xno != "x$with_magick_pp"],
   AS_CASE("x$with_magick_pp",
     [xGraphicsMagick],
       [PKG_CHECK_MODULES([LIBMAGICK_PP], [GraphicsMagick++],
          [AC_DEFINE([HAVE_GRAPHICS_MAGICK_PP], [1])])
       ],
     [xImageMagick],
       [PKG_CHECK_MODULES([LIBMAGICK_PP], [Magick++ > 6.4.2],
          [AC_DEFINE([HAVE_IMAGE_MAGICK_PP], [1])])
       ],
     [xyes|xcheck],
       [PKG_CHECK_MODULES([LIBMAGICK_PP], [GraphicsMagick++],
          [AC_DEFINE([HAVE_GRAPHICS_MAGICK_PP], [1])],
          [PKG_CHECK_MODULES([LIBMAGICK_PP], [Magick++ > 6.4.2],
             [AC_DEFINE([HAVE_IMAGE_MAGICK_PP], [1])],
             [AS_IF([test xcheck != "x$with_magick_pp"],
                [AC_MSG_ERROR([magick-pp required but not found])])
             ])
          ])
       ],
     [dnl default case
      AC_MSG_ERROR([unknown value: --with-magick-pp=$with_magick_pp])
     ]))
AM_CONDITIONAL([have_libmagick_pp], [test x != "x$LIBMAGICK_PP_LIBS"])

AS_IF([test xno != "x$with_magick"],
   AS_CASE("x$with_magick",
     [xGraphicsMagick],
       [AC_CHECK_PROGS([MAGICK_CONVERT], [gm])
        AS_IF([test xgm != x$MAGICK_CONVERT],
          [AC_MSG_ERROR([$with_magick requested but not found])])
        AC_DEFINE([HAVE_GRAPHICS_MAGICK], [1])
        MAGICK_CONVERT="gm convert"
       ],
     [xImageMagick],
       [AC_CHECK_PROGS([MAGICK_CONVERT], [convert])
        AS_IF([test xconvert != x$MAGICK_CONVERT],
          [AC_MSG_ERROR([$with_magick requested but not found])])
        AC_DEFINE([HAVE_IMAGE_MAGICK], [1])
       ],
     [xyes|xcheck],
       [AC_CHECK_PROGS([MAGICK_CONVERT], [gm convert])
        AS_CASE(x$MAGICK_CONVERT,
          [xgm],
            [AC_DEFINE([HAVE_GRAPHICS_MAGICK], [1])
             MAGICK_CONVERT="gm convert"
            ],
          [xconvert],
            [AC_DEFINE([HAVE_IMAGE_MAGICK], [1])
            ],
          [dnl default case
           AS_IF([test xcheck != "x$with_magick"],
             [AC_MSG_ERROR([magick required but not found])])
          ])
       ],
     [dnl default case
      AC_MSG_ERROR([unknown value: --with-magick=$with_magick])
     ]))
AC_DEFINE_UNQUOTED([MAGICK_CONVERT], ["$MAGICK_CONVERT"])
AM_CONDITIONAL([have_magick], [test x != "x$MAGICK_CONVERT"])
```

This means that if arguments are not provided for `--with-magick` and `--with-magick-pp` options, then automatic detection is performed, with GraphicsMagick checked first and ImageMagick checked second.
Please use the following in ebuild:
```
--with-magick=$(usex graphicsmagick GraphicsMagick ImageMagick)
--with-magick-pp=$(usex graphicsmagick GraphicsMagick ImageMagick)
```

Reply:
Unfortunately iscan fails when built without "magick" support as mentioned earlier (including bug report), so effectively it becomes a hard dependency.

Reply:
See my comment above. `graphicsmagick` USE flag is not used to enable/disable support for *Magick, but to select one of them instead of the other one.

Reply:
I see now.